src/Service.Tests/OpenApiDocumentor/ParameterValidationTests.cs

```diff
@@ -0,0 +1,240 @@
+
+        Assert.AreEqual(2, openApiDocument.Paths.Count);
+        Assert.IsTrue(openApiDocument.Paths.ContainsKey($"/{entityName}"));
+        Assert.IsTrue(openApiDocument.Paths.ContainsKey($"/{entityName}/id/{{id}}"));
+        foreach ((string pathName, OpenApiPathItem pathItem) in openApiDocument.Paths)
+        {
+            string apiPathWithParam = $"/{entityName}/id/{{id}}";
+            if (pathName.Equals(apiPathWithParam))
+            {
+                Assert.IsTrue(pathItem.Parameters.Count is 1);
+                OpenApiParameter pathParameter = pathItem.Parameters.First();
+                Assert.AreEqual("id", pathParameter.Name);
+                Assert.AreEqual(ParameterLocation.Path, pathParameter.In);
+                Assert.IsTrue(pathParameter.Required);
+            }
+            else
+            {
+                // Get All and POST method with path /entityName, will have no path parameters.
+                Assert.IsTrue(pathItem.Parameters.Count is 0);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Validates that the default set of query parameters are generated for GET methods in Table/Views.
+    /// $select, $filter, $orderby, $first, $after are the query parameters.
+    /// </summary>
+    /// <param name="entityName">The name of the entity.</param>
+    /// <param name="objectName">The name of the database object.</param>
+    /// <param name="entitySourceType">The source type of the entity.</param>
+    [DataTestMethod]
+    [DataRow("BooksTable", "books", EntitySourceType.Table, DisplayName = "Table with query parameters")]
+    [DataRow("BooksView", "books_view_all", EntitySourceType.View, DisplayName = "View with query parameters")]
+    public async Task TestQueryParametersAddedForGEToperationOnTablesAndViews(string entityName, string objectName, EntitySourceType entitySourceType)
+    {
+        EntitySource entitySource = new(Object: objectName, entitySourceType, null, null);
+        OpenApiDocument openApiDocument = await GenerateOpenApiDocumentForGivenEntityAsync(entityName, entitySource);
+
+        Assert.AreEqual(2, openApiDocument.Paths.Count);
+        Assert.IsTrue(openApiDocument.Paths.ContainsKey($"/{entityName}"));
+        Assert.IsTrue(openApiDocument.Paths.ContainsKey($"/{entityName}/id/{{id}}"));
+
+        // Assert that path without id has all the query parameters.
+        Assert.IsTrue(openApiDocument.Paths[$"/{entityName}"].Operations.ContainsKey(OperationType.Get));
+        Assert.AreEqual(5, openApiDocument.Paths[$"/{entityName}"].Operations[OperationType.Get].Parameters.Count);
+
+        // Asserting on all the parameters
+        List<OpenApiParameter> openApiParameters = openApiDocument.Paths[$"/{entityName}"].Operations[OperationType.Get].Parameters.ToList();
+        Assert.IsTrue(openApiParameters.Any(param => param.In is ParameterLocation.Query && param.Name.Equals("$select") && param.Schema.Type.Equals("string")));
```

Comment:
do we have defined constants for all the query parameters $select, $filter etc?

Reply:
added. Thanks for pointing out